tests/Semaphore002.hs

```diff
@@ -0,0 +1,15 @@
+module Main (main) where
+
+import Control.Concurrent
+import System.Posix
+
+main :: IO ()
+main = do
+  sem <- semOpen "/test" OpenSemFlags {semCreate = True, semExclusive = False} stdFileMode 0
+  forkIO $ do
```

Comment:
This triggers a warning "A do-notation statement discarded a result of type ‘ThreadId’".